CMakeLists.txt

```diff
@@ -1,81 +1,146 @@
 set_property ( CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS ${CMAKE_CONFIGURATION_TYPES} )
 
-#Name project and specify source languages
-project(coarray_icar VERSION 0.1.0 LANGUAGES Fortran)
+# Name project and specify source languages
+project(morfeus VERSION "0.1" LANGUAGES Fortran)
 
 #Print an error message on an attempt to build inside the source directory tree:
 if ("${CMAKE_CURRENT_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
   message(FATAL_ERROR "ERROR! "
     "CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}"
     " == CMAKE_CURRENT_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}"
     "\nThis archive does not support in-source builds:\n"
-    "You must now delete the CMakeCache.txt file and the CMakeFiles/ directory under"
+    "You must now delete the CMakeCache.txt file and the CMakeFiles/ directory under "
     "the 'src' source directory or you will not be able to configure correctly!"
     "\nYou must now run something like:\n"
     "  $ rm -r CMakeCache.txt CMakeFiles/"
     "\n"
     "Please create a directory outside the opencoarrays source tree and build under that outside directory "
     "in a manner such as\n"
-    "  $ mkdir build-opencarrays\n"
-    "  $ cd build-opencoarrays\n"
-    "  $ CC=gcc FC=gfortran cmake <path-to-opencoarrays-source-directory> -DCMAKE_INSTALL_PREFIX=<path-to-install-directory>\n"
+    "  $ mkdir -p coarray_icar/build\n"
+    "  $ cd coarray_icar/build\n"
+    "  $ FC=caf cmake ..\n"
     "\nsubstituting the appropriate syntax for your shell (the above line assumes the bash shell)."
     )
 endif()
 
-#Report untested Fortran compiler unless explicitly directed to build all examples.
-if ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU" )
-  set(gfortran_compiler true)
-  set ( CMAKE_Fortran_FLAGS_CODECOVERAGE "-fprofile-arcs -ftest-coverage -O0"
-    CACHE STRING "Code coverage C compiler flags")
-elseif ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "Intel" )
-  set(intel_compiler true)
-  set ( CMAKE_Fortran_FLAGS "-coarray")
-else()
+set (CMAKE_Fortran_FLAGS_DEBUG "-g -DUSE_ASSERTIONS=.true." CACHE
+  STRING "Flags used by the compiler during debug builds."
+  FORCE)
+set (CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-g -O2 -DNDEBUG -DUSE_ASSERTIONS=.true." CACHE
+  STRING "Flags used by the compiler during release builds with debug info."
+  FORCE)
+
+if (NOT "${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU|Intel|Cray" )
   message(WARNING
     "\n"
     "Attempting to build with untested Fortran compiler: ${CMAKE_Fortran_COMPILER_ID}. "
-    "Please report any failures via https://github.com/rouson/coarray_icar/issues\n\n"
+    "Please report any failures at https://github.com/gutmann/coarray_icar/issues\n\n"
   )
 endif()
 
-if(gfortran_compiler)
-  set(CMAKE_REQUIRED_FLAGS "-fcoarray=single -ffree-form") # set temporary flag
+if ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU" )
+  set(coarray_flags "-fcoarray=single")
+elseif ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "Intel" )
+  set(coarray_flags "-coarray=shared")
```

Comment:
Can this set `-coarray=distributed` somewhere?

Reply:
Good point.  I suppose I should add a flag for the user to set this.  `shared` is a good default because `distributed` is only available if the user has Intel Parallel Studio Cluster Edition, but we should give the option to switch to `distributed`.  I'll add that to the pull request.